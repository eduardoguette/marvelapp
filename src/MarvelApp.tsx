import { Header, Main } from '@/layout'
import { Outlet } from 'react-router-dom'


export default function MarvelApp() {
  return (
    <> 
      <Header/>
      <Main>
        <Outlet />
      </Main>
    </>
  )
}
