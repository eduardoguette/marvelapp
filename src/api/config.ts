import axios from 'axios'

  const defaults = {
  baseUrl: 'https://gateway.marvel.com:443/v1/public/',
  queryParams: {
    apikey: '54d2294be5f84e38f732119e45c34b88',
    limit: 50,
    offset: 0
  }
}

export const marvelappApi = axios.create({
  baseURL: defaults.baseUrl,
  headers: {
    'Content-Type': 'application/json'
  },
  params: defaults.queryParams
})
