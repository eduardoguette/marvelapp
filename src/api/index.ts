export * from './characters'
export * from './config'

