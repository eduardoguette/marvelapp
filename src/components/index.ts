export * from './Carrousel'
export * from './Characters/Card'
export * from './Commics'
export * from './commons'

