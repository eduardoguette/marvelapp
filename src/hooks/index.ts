export * from './useDebounce'
export * from './useFetchCharacters'
