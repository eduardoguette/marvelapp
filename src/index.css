:root {
  /* Colores */
  --primary: #ec1d24;
  --secondary: #000000;
  --gray: #333333;
  --white: #ffffff;

  /* fonts */
  --font-basic: Roboto Condense, system-ui, -apple-system, BlinkMacSystemFont,
    'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue',
    sans-serif;

  /* body font */
  --body: normal 16px / 24px var(--font-basic);

  /* headline fonts */
  --headline1: normal 700 40px / 46.88px var(--font-basic);
  --headline2: normal 700 32px / 37.5px var(--font-basic);
  --headline3: normal 700 24px / 28.13px var(--font-basic);
  --headline4: normal 700 16px / 24px var(--font-basic);

  /* caption font */
  --caption: normal 12px / 14.06px var(--font-basic);
}

@font-face {
  font-family: Roboto Condense;
  src: url('./assets/fonts/RobotoCondensed-Regular.ttf') format('truetype');
  font-weight: normal;
  font-style: normal;
}

@font-face {
  font-family: Roboto Condense;
  src: url('./assets/fonts/RobotoCondensed-Bold.ttf') format('truetype');
  font-weight: 700;
  font-style: normal;
}

*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font: var(--body);
}

li {
  list-style: none;
}
