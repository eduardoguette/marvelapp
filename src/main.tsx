import React from 'react'
import ReactDOM from 'react-dom/client'
import { RouterProvider } from 'react-router-dom'

import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { AppProvider } from './context'
import './index.css'
import { router } from './routes/Root'
const queryClient = new QueryClient()
ReactDOM.createRoot(document.getElementById('root')!).render(
  <React.StrictMode>
    <AppProvider>
      <QueryClientProvider client={queryClient}>
        <RouterProvider router={router} />
      </QueryClientProvider>
    </AppProvider>
  </React.StrictMode>
)
