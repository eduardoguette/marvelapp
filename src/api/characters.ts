import { marvelappApi } from './config'

export const defaults = {
  getAllCharacters: '/characters'
}

 
export const characters = {
  getAll: () => marvelappApi.get(defaults.getAllCharacters),
  getById: (id: string) => marvelappApi.get(`${defaults.getAllCharacters}/${id}`),
  getCommics: (id: string) => marvelappApi.get(`${defaults.getAllCharacters}/${id}/comics`)
}
