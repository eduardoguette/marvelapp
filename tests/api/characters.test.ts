import { Mock, describe, expect, it, vi } from 'vitest'
import { defaults } from '../../src/api'
import { characters } from '../../src/api/characters'
import { marvelappApi } from '../../src/api/config'

// Mockear marvelappApi
vi.mock('../../src/api/config.ts', () => ({
  marvelappApi: {
    get: vi.fn()
  }
}))

describe('characters API', () => {
  it('getAll debería llamar a marvelappApi.get con la URL correcta y coincidir con el snapshot', async () => {
    const mockResponse = { data: { results: [] } }

    ;(marvelappApi.get as unknown as Mock).mockResolvedValue(mockResponse)

    const response = await characters.getAll()
    expect(marvelappApi.get).toHaveBeenCalledWith(defaults.getAllCharacters)
    expect(response).toMatchSnapshot()
  })
  it('getById debería llamar a marvelappApi.get con la URL correcta y coincidir con el snapshot', async () => {
    const mockResponse = {
      data: { results: [{ id: '1', name: 'Spider-Man' }] }
    }

    ;(marvelappApi.get as unknown as Mock).mockResolvedValue(mockResponse)

    const id = '1'
    const response = await characters.getById(id)

    expect(marvelappApi.get).toHaveBeenCalledWith(
      `${defaults.getAllCharacters}/${id}`
    )
    expect(response.data.results[0].id).toBe('1')
    expect(Array.isArray(response.data.results)).toBe(true)
    expect(response).toMatchSnapshot()
  })

  it('getCommics debería llamar a marvelappApi.get con la URL correcta y coincidir con el snapshot', async () => {
    const mockResponse = { data: { results: [] } }

    ;(marvelappApi.get as unknown as Mock).mockResolvedValue(mockResponse)

    const id = '1'
    const response = await characters.getCommics(id)
    expect(marvelappApi.get).toHaveBeenCalledWith(
      `${defaults.getAllCharacters}/${id}/comics`
    )
    expect(Array.isArray(response.data.results)).toBe(true)
    expect(response).toMatchSnapshot()
  })
})
