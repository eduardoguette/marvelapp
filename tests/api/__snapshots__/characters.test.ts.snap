// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`characters API > getAll debería llamar a marvelappApi.get con la URL correcta y coincidir con el snapshot 1`] = `
{
  "data": {
    "results": [],
  },
}
`;

exports[`characters API > getById debería llamar a marvelappApi.get con la URL correcta y coincidir con el snapshot 1`] = `
{
  "data": {
    "results": [
      {
        "id": "1",
        "name": "Spider-Man",
      },
    ],
  },
}
`;

exports[`characters API > getCommics debería llamar a marvelappApi.get con la URL correcta y coincidir con el snapshot 1`] = `
{
  "data": {
    "results": [],
  },
}
`;
